src/plugins/intel_gpu/include/intel_gpu/plugin/remote_context.hpp

```diff
@@ -476,6 +476,11 @@ class TypedExecutionContext : public TpublicContextAPI {
                                    const Config& config = {})
         : _impl(plugin, params, config) {}
 
+    ~TypedExecutionContext() {
+        shared_surf_reg.clear();
+        shared_obj_reg.clear();
```

Comment:
How does it fix the issue: looks like member of class should be destroyed in class destructor automatically. is it order of destruction is important?
If answer is yes, then the most simplest solution is to change order of class member declarations

```
std::map<surf_key, InferenceEngine::RemoteBlob::Ptr> shared_surf_reg;
std::map<cldnn::shared_handle, InferenceEngine::RemoteBlob::Ptr> shared_obj_reg; 
```
on

```
std::map<cldnn::shared_handle, InferenceEngine::RemoteBlob::Ptr> shared_obj_reg; 
std::map<surf_key, InferenceEngine::RemoteBlob::Ptr> shared_surf_reg;
```
and use natural default destructor

Reply:
Yes, the order of destructors is important here.
We have to guarantee that those two maps, which are shared blobs cache, are destroyed before _impl object, which holds references to DX11 handles.
To make the solution independent from any future class design changes, we destroy the cache maps in destructor, and then the rest is destroyed automatically. This way we don't have to care about declaration order or any other factors that might interfere.

Reply:
thanks for your clarifications, it makes sense